src/parthenon_manager.cpp

```diff
@@ -77,6 +77,17 @@ ParthenonStatus ParthenonManager::ParthenonInit(int argc, char *argv[]) {
     return ParthenonStatus::complete;
   }
 
+  // Allow for user overrides to default Parthenon functions
+  if (finput->SetFillDerivedFunctions != nullptr) {
+    SetFillDerivedFunctions = finput->SetFillDerivedFunctions;
+  }
+  if (finput->ProcessProperties != nullptr) {
+    ProcessProperties = finput->ProcessProperties;
+  }
+  if (finput->ProcessPackages != nullptr) {
+    ProcessPackages = finput->ProcessPackages;
+  }
+
```

Comment:
hmm. Could we just have the defaults be explicitly set in the fields of the `finput` struct when it's defined? Currently they're defaulted to `nullptr` but we could use the explicit defaults we want maybe?

Reply:
As of now the defaults are member functions of their respective classes (`Mesh` etc.) and since at least `ParthenonManager` includes the `FunctionInput` class, I think you may get in trouble with circular dependencies trying switch to this. I think that would work but it would only save you so many lines (and classes like `Mesh` would spend at least some time with null function pointers whereas right now they are set in the class declaration to the defaults)

Reply:
hmm ok fair enough